### PR TITLE
net: Finish FIONBIO default action if si_ioctl return OK

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

Continue the work: https://github.com/apache/nuttx/pull/6976. https://github.com/apache/nuttx/pull/8728 contain additional information.

## Impact

Handle FIONBIO more correctly for socket case

### CODE REVIEW

File: net/netdev/netdev_ioctl.c [hunk 1]
```diff
@@ -1491,72 +1501,41 @@ static int netdev_file_ioctl(FAR struct socket *psock, int cmd,
...
             }
           else
             {
               nerr("ERROR: Non-blocking not supported for this socket\n");
               ret = -ENOSYS;
             }
         }
-        break;
 
-      default:
```
Comment (pkarashchenko) [inline net/netdev/netdev_ioctl.c:1522]: Let's keep `default: break;`?

**Review by pkarashchenko — APPROVED** [review]

File: net/netdev/netdev_ioctl.c (hunk 1 above)
Comment (xiaoxiang781216) [reply net/netdev/netdev_ioctl.c:1522]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Comment by jturnsek** [comment]
It seems to work.

**Pull request merged**